tests/ashley/machina_extensions/forum_conversation/test_forum_conversation.py

```diff
@@ -37,162 +36,423 @@ def setUp(self):
+        topic = TopicFactory(forum=self.forum, poster=user1)
+        # Add two posts for topic
+        PostFactory(topic=topic, poster=user1)
+        PostFactory(topic=topic, poster=user2)
+        form = PostForm(user=user3, forum=self.forum, topic=topic)
+        # Two users should be listed
         assert form.fields["content"].widget.attrs["mention_users"] == [
-            {"name": "Alfred", "link": f"/forum/member/profile/{user2.id}/"},
-            {"name": "Benoit", "link": f"/forum/member/profile/{user1.id}/"},
+            {
+                "name": "Alfred",
+                "link": f"/forum/member/profile/{user2.id}/",
+                "username": "user2",
+            },
+            {
+                "name": "Benoit",
+                "link": f"/forum/member/profile/{user1.id}/",
+                "username": "user1",
+            },
         ]
 
         # Set up new topic and initial post with user4
-        topic2 = TopicFactory(forum=self.forum, poster=user4)
-        post_topic2_user4 = PostFactory(subject="Réponse", topic=topic2, poster=user4)
+        topic2 = TopicFactory(forum=self.forum, poster=user3)
+        PostFactory(topic=topic2, poster=user3)
 
-        # user3 loads the form for topic2
-        form = PostForm(data=form_data, user=user3, forum=self.forum, topic=topic2)
+        # user1 loads the form for topic2
+        form = PostForm(
+            user=user1,
+            forum=self.forum,
+            topic=topic2,
+        )
         assert form.fields["content"].widget.attrs["mention_users"] == [
             {
-                "name": "user4",
-                "link": f"/forum/member/profile/{user4.id}/",
+                "name": "Aurélien",
+                "link": f"/forum/member/profile/{user3.id}/",
+                "username": "user3",
             }
         ]
-        # user3 loads topic, nothing should have changed as user4 only posted in another topic
-        form = PostForm(data=form_data, user=user3, forum=self.forum, topic=topic)
+        # user3 loads topic, nothing should have changed as user3 only posted in another topic
+        form = PostForm(
+            user=user3,
+            forum=self.forum,
+            topic=topic,
+        )
+        assert form.fields["content"].widget.attrs["mention_users"] == [
+            {
+                "name": "Alfred",
+                "link": f"/forum/member/profile/{user2.id}/",
+                "username": "user2",
+            },
+            {
+                "name": "Benoit",
+                "link": f"/forum/member/profile/{user1.id}/",
+                "username": "user1",
+            },
+        ]
+
+    def test_list_active_users_only_concerns_users_with_approved_posts(self):
+        """
+        The form loads the list of active users for the current topic. We control that the list
+        only concerns users that have approved posts
+        """
+        user1 = UserFactory(
+            email="user1@mooc.fr",
```

Comment:
Why do you define an email address explicitly instead of letting the factory generate it?
(this comments apply for all your User instances in your tests).

In addition to that, I do not recommend to use a domain name that belong to someone else, it can lead to hazardous situations. I recommend to use domains reserved for test and documentation like `example.com` (see [IANA reserved domains](https://www.iana.org/domains/reserved) or [RFC2606](https://tools.ietf.org/html/rfc2606#section-2))

Reply:
really good comment ! no point ! thank you